### PR TITLE
make pipeline cleaner

### PR DESCRIPTION
**Pull request by johnnychen94**
The first commit flattens the nested code structure by removing the `try-catch` block. There're no function changes except using `isfile` to check if reference file exists. (Is there any special reason using `try-catch` here?)

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -79,44 +79,44 @@ end
...
-                else
-                    @test false
-                end
-            else
-                error("You need to run the tests interactively with 'include(\"test/runtests.jl\")' to create new reference images")
-            end
+
+        return nothing # skip current test case
+    end
+
```
Comment (oxinabox) [inline src/core.jl]:
```suggestion
    # File exists:
```

**Comment by oxinabox** [comment]
Nice, good improvement.
Can also split the content of the two branchs of the `if` into their own functions.
One for handing non-existant file (`handle_nonextistant` ?), and one for handling disagreeing contents (`handle_disagreement` ?).

**Comment by johnnychen94** [comment]
Mind to merge this? I think I'd like to open a new PR so that it's easier to be reviewed.

**Comment by oxinabox** [comment]
I stand by my above comments re breaking it into more functions.

**Comment by johnnychen94** [comment]
Ahh, yes, but I have"better" ideas than simply breaking it into parts, and that would change the code structure.

**Comment by johnnychen94** [comment]
Or I could push commits into this branch if you like.

**Comment by oxinabox** [comment]
Ok lets merge this first.

**Pull request merged**